core/server/src/main/java/alluxio/worker/block/evictor/LIRSEvictor.java

```diff
@@ -0,0 +1,573 @@
+  private static final float LINKED_HASH_MAP_INIT_LOAD_FACTOR = 0.75f;
+  private static final boolean LINKED_HASH_MAP_ACCESS_ORDERED = false;
+  private static final boolean UNUSED_MAP_VALUE = true;
+
+  // The percent of HIR blocks on each tier
+  private final double mHIRPercent;
+  // The percent of LIR blocks on each tier
+  private final double mLIRPercent;
+  private final Configuration mConfiguration;
+  // Map from the location of a StorageDir to its condition of used space
+  private Map<BlockStoreLocation, SpaceContainer> mSpaceManager =
+      new ConcurrentHashMap<BlockStoreLocation, SpaceContainer>();
+  /**
+   * Map from the the block id to the block size. Maintain the information of block size is because
+   * the space of HIR block or LIR block need to be reclaimed when
+   * {@link #onRemoveBlockByClient(long, long)} or {@link #onRemoveBlockByWorker(long, long)} is
+   * called.
+   */
+  private Map<Long, Long> mBlockIdToSize = new ConcurrentHashMap<Long, Long>();
+  /**
+   * Record the information of all LIR blocks. Pair<BlockStoreLocation, Long> identifies a block
+   * because a block may contain more than one ghost blocks on all tiers.
+   */
+  private Map<Pair<BlockStoreLocation, Long>, BlockLIRSInfo> mLIRCache =
+      Collections.synchronizedMap(new LinkedHashMap<Pair<BlockStoreLocation, Long>, BlockLIRSInfo>(
+          LINKED_HASH_MAP_INIT_CAPACITY, LINKED_HASH_MAP_INIT_LOAD_FACTOR,
+          LINKED_HASH_MAP_ACCESS_ORDERED));
+  // Record the information of all HIR blocks.
+  private Map<Pair<BlockStoreLocation, Long>, BlockLIRSInfo> mHIRCache =
+      Collections.synchronizedMap(new LinkedHashMap<Pair<BlockStoreLocation, Long>, BlockLIRSInfo>(
+          LINKED_HASH_MAP_INIT_CAPACITY, LINKED_HASH_MAP_INIT_LOAD_FACTOR,
+          LINKED_HASH_MAP_ACCESS_ORDERED));
+  /**
+   * Record the information of all temp blocks. Temp blocks are blocks moved from other tiers and
+   * still not accessed by any clients.
+   */
+  private Map<Pair<BlockStoreLocation, Long>, Boolean> mTmpCache = Collections.synchronizedMap(
+      new LinkedHashMap<Pair<BlockStoreLocation, Long>, Boolean>(LINKED_HASH_MAP_INIT_CAPACITY,
+          LINKED_HASH_MAP_INIT_LOAD_FACTOR, LINKED_HASH_MAP_ACCESS_ORDERED));
+
+  /**
+   * Create a new instance of {@link LIRSEvictor}.
+   *
+   * @param view a view of block metadata information
+   * @param allocator an allocation policy
+   */
+  public LIRSEvictor(BlockMetadataManagerView view, Allocator allocator) {
+    super(view, allocator);
+    mConfiguration = WorkerContext.getConf();
+    mHIRPercent = mConfiguration.getDouble(Constants.WORKER_EVICTOR_LIRS_HIR_PERCENT);
+    Preconditions.checkArgument(mHIRPercent >= 0 && mHIRPercent <= 1,
+        "HIR percent should be larger than 0 and less than 1");
+    mLIRPercent = 1.0 - mHIRPercent;
+
+    for (StorageTierView tier : view.getTierViews()) {
+      for (StorageDirView dir : tier.getDirViews()) {

```

Comment:
@uronce-cc You are right. Actually, the tier still can be thought as comprising 2 parts logically.